core/src/message/context.rs

```diff
@@ -352,10 +345,18 @@ impl MessageContext {
 
     /// Pushes payload into stored payload by handle.
     pub fn send_push(&mut self, handle: u32, buffer: &[u8]) -> Result<(), Error> {
+        let bytes_amount = u32::try_from(buffer.len()).map_err(|_| Error::MaxMessageSizeExceed)?;
```

Comment:
Actually both `OutgoingMessagesBytesLimitExceeded` and  `MaxMessageSizeExceed` are suitable to be returned. But I chose the old one to be more suitable for legacy